src/adapter/src/catalog.rs

```diff
@@ -5815,7 +5817,9 @@ impl Catalog {
         create_sql: String,
         pcx: Option<&PlanContext>,
     ) -> Result<CatalogItem, AdapterError> {
-        let session_catalog = self.for_system_session();
+        let mut session_catalog = self.for_system_session();
+        enable_features_required_for_catalog_open(&mut session_catalog);
```

Comment:
Doesn't this mean that the feature gets turned on for every environment, every time it starts up?

Reply:
Yes, but AFAICT the `session_catalog` constructed here is only used in the
- `mz_sql::parse::parse`,
- `mz_sql::names::resolve`, and 
- `mz_sql::plan::plan`
calls of the enclosing function (`parse_item` / `parse_view_item`) and then destroyed, so this override should only affect the scope of `Catalog::open`. For regular client connections the original value should be still be used.

Reply:
But the system variables are scoped to the entire catalog, not just an individual session. So doesn't this change the value for all future sessions too?

Reply:
Perhaps we should only enable it if it's disabled and then after re-disable it.

Reply:
> Perhaps we should only enable it if it's disabled

I am only doing this in order to avoid cloning if not needed (see the guards in `enable_features_required_for_catalog_open`). It will still affect most `environmentd` instances in production, though...

> and then after re-disable it.

I'm not sure if this is needed if we are going to drop the clone either way at the end of the method. It's more future proof but then I need two methods instead of one and I need to around the set of features that need to be flipped as state.